### ISSUE TITLE
Annotation for vCOCO dataset

### ISSUE THREAD

**Issue by BozhouZha** [YueLiao/PPDM#16]
Hi there,

Amazing work you've done! I have finished reproducing similar mAP via your repository and that's amazingly well. For the next step, I plan to try out your methodology on v-COCO dataset, however I note that you refer to iCAN project for the vcoco part. I'm curious that whether you prepared the annotation for the COCO dataset part?

I navigated to the iCAN project webpage, but found the annotations that they provided is not quite of "json" format and the syntax of which the vcoco_eval cannot even parse.

If you can help, I appreciate it a lot!

**Comment by YueLiao** [comment]
Thanks for your interest in our work. This is the [script](https://gist.github.com/YueLiao/9fc76ec66ea04c97fb1322f3ee735a0a) which we used to get the iCAN annotations from V-COCO. But we didn't check the script carefully, you'd better double-check it.

**Comment by BozhouZha** [comment]
> Thanks for your interest in our work. This is the [script](https://gist.github.com/YueLiao/9fc76ec66ea04c97fb1322f3ee735a0a) which we used to get the iCAN annotations from V-COCO. But we didn't check the script carefully, you'd better double-check it.

That's great! Thank you very much! Good luck on your future work!

**Comment by BestSongEver** [comment]
@BozhouZha  
Hi, sorry to bother u.
Following the  above suggestions, i can generate "verbcoco_trainval.json" (using " Trainval_GT_VCOCO.pkl " and "instances_vcoco_all_2014.json").
But i wonder how i can generate "verbcoco_test.json" (while we do not have a "Test_GT_VCOCO.pkl" from iCAN) ?

Thank you very much for your reply and help!

**Comment by BozhouZha** [comment]
> @BozhouZha
> Hi, sorry to bother u.
> Following the above suggestions, i can generate "verbcoco_trainval.json" (using " Trainval_GT_VCOCO.pkl " and "instances_vcoco_all_2014.json"), which is file with size of 2,076 kb ?
> But i wonder how i can generate "verbcoco_test.json" (while we do not have a "Test_GT_VCOCO.pkl" from iCAN) ?
> 
> Thank you very much for your reply and help!

Sorry for my late responce. That's actually a good point. In fact, even for the Trainval_GT_VCOCO.pkl, I don't think they (iCAN) did cover both the testing set and validating set. You can check the number of instances of Trainval_GT_VCOCO.pkl, which is far less than the summation of test and val. Besides, the mAP is not as good as the trial on HICO. I currently gave up the attempt to reimplement on VCOCO, so I'm so sorry for not providing you with useful info. You might as well turn to YueLiao for help directly, he is super kind and helpful.

**Comment by BestSongEver** [comment]
As mentioned above, following ur suggestions, i can generate "verbcoco_trainval.json" (using " Trainval_GT_VCOCO.pkl " and "instances_vcoco_all_2014.json").
But i wonder how i can generate "verbcoco_test.json" (while we do not have a "Test_GT_VCOCO.pkl" from iCAN) ?
Could u please offer me some help?  or ur generated "verbcoco_test.json" ? @YueLiao 
Thank you very very much for your reply and help!

**Comment by YueLiao** [comment]
> As mentioned above, following ur suggestions, i can generate "verbcoco_trainval.json" (using " Trainval_GT_VCOCO.pkl " and "instances_vcoco_all_2014.json").
> But i wonder how i can generate "verbcoco_test.json" (while we do not have a "Test_GT_VCOCO.pkl" from iCAN) ?
> Could u please offer me some help? or ur generated "verbcoco_test.json" ? @YueLiao
> Thank you very very much for your reply and help!

You can generate the "verbcoco_test.json" from the official annotation by taking [this script](https://gist.github.com/YueLiao/916a1868c4efeda79477be8cb3f9df59) as a reference, but I cannot make sure for its correctness, which is only a "toy" script.

**Comment by BestSongEver** [comment]
@YueLiao 
Thanx for ur lovely help! With ur script, i can run the training and testing process on VCOCO.
However, with the 'dla_34' model and ur script, i get the mAP of only : 0.06808124378457435   max recall: 0.1751444416169466
I'd appreciate it if you could give me some advice.
Wish you all the best !

**Comment by lylinyang** [comment]
> @YueLiao
> Thanx for ur lovely help! With ur script, i can run the training and testing process on VCOCO.
> However, with the 'dla_34' model and ur script, i get the mAP of only : 0.06808124378457435 max recall: 0.1751444416169466
> I'd appreciate it if you could give me some advice.
> Wish you all the best !
@YueLiao 
With the resnet18 model, we also get the mAP on VCOCO which is only : mAP: 0.05442602959254834   max recall: 0.1533439089945431.
Any advices please? Hope to

**Comment by YueLiao** [comment]
> > @YueLiao
> > Thanx for ur lovely help! With ur script, i can run the training and testing process on VCOCO.
> > However, with the 'dla_34' model and ur script, i get the mAP of only : 0.06808124378457435 max recall: 0.1751444416169466
> > I'd appreciate it if you could give me some advice.
> > Wish you all the best !
> > @YueLiao
> > With the resnet18 model, we also get the mAP on VCOCO which is only : mAP: 0.05442602959254834   max recall: 0.1533439089945431.
> > Any advices please? Look forward to your kind reply.

Could you give more details, e.g., pre-trained models, inference process?

**Comment by BestSongEver** [comment]
> > > @YueLiao
> > > Thanx for ur lovely help! With ur script, i can run the training and testing process on VCOCO.
> > > However, with the 'dla_34' model and ur script, i get the mAP of only : 0.06808124378457435 max recall: 0.1751444416169466
> > > I'd appreciate it if you could give me some advice.
> > > Wish you all the best !
> > > @YueLiao
> > > With the resnet18 model, we also get the mAP on VCOCO which is only : mAP: 0.05442602959254834   max recall: 0.1533439089945431.
> > > Any advices please? Look forward to your kind reply.
> 
> Could you provide more details, e.g., pre-trained models, inference processes?

My training script onV-COCO is:
python -u  main.py hoidet --exp_id hoidet_vcoco_dla --batch_size 15 --master_batch 2  --lr 4.5e-4 --gpus 0 --num_workers 2 --val_intervals 100000 --image_dir images --load_model ../models/ctdet_coco_dla_2x.pth --dataset vcoco --root_path '/userhome/project/PPDM/Dataset'

As above, i trained the  'dla_34' model with a small batch_size on a single GPU, and i did not change any other configuration & parameter & code (e.g.,  pre-trained models, inference processes). 
By the way, we generated "verbcoco_trainval.json", "verbcoco_test.json" and "corre_vcoco.npy" with the scripts you provided for training and testing.
Thanx for your advice.

**Comment by lylinyang** [comment]
Thinks for your reply. The pre-trained model is "ctdet_coco_resdcn18.pth", and we follow your script to prepare the VCOCO data for inference. I attached the training log for you, please check it if we make something wrong:
[log_train_ori_vcoco_resnet_18_0724.log](https://github.com/YueLiao/PPDM/files/4979724/log_train_ori_vcoco_resnet_18_0724.log)

**Comment by lylinyang** [comment]
> Thinks for your reply. The pre-trained model is "ctdet_coco_resdcn18.pth", and we follow your script to prepare the VCOCO data for inference. I attached the training log for you, please check it if we make something wrong:
> [log_train_ori_vcoco_resnet_18_0724.log](https://github.com/YueLiao/PPDM/files/4979724/log_train_ori_vcoco_resnet_18_0724.log)

The train and test script is as follows:

nohup python main.py  hoidet --batch_size 256 --master_batch 16 --lr 4.5e-4 --gpus 0,1,2,3,4,5,6,7  --num_workers 16  --load_model ../models/ctdet_coco_resdcn18.pth --arch resdcn_18 --image_dir images/train2014 --dataset vcoco --exp_id hoidet_vcoco_resdcn18  --print_iter 1 >> log_train_ori_vcoco_resnet_18_0724.log 2>&1 &

python test_hoi.py hoidet --exp_id hoidet_vcoco_resdcn18 --arch resdcn_18 --load_model ../exp/hoidet/hoidet_vcoco_resdcn18/model_140.pth --gpus 0 --dataset vcoco --image_dir images/val2014 --test_with_eval

**Comment by YueLiao** [comment]
> > Thinks for your reply. The pre-trained model is "ctdet_coco_resdcn18.pth", and we follow your script to prepare the VCOCO data for inference. I attached the training log for you, please check it if we make something wrong:
> > [log_train_ori_vcoco_resnet_18_0724.log](https://github.com/YueLiao/PPDM/files/4979724/log_train_ori_vcoco_resnet_18_0724.log)
> 
> The train and test script is as follows:
> 
> nohup python main.py hoidet --batch_size 256 --master_batch 16 --lr 4.5e-4 --gpus 0,1,2,3,4,5,6,7 --num_workers 16 --load_model ../models/ctdet_coco_resdcn18.pth --arch resdcn_18 --image_dir images/train2014 --dataset vcoco --exp_id hoidet_vcoco_resdcn18 --print_iter 1 >> log_train_ori_vcoco_resnet_18_0724.log 2>&1 &
> 
> python test_hoi.py hoidet --exp_id hoidet_vcoco_resdcn18 --arch resdcn_18 --load_model ../exp/hoidet/hoidet_vcoco_resdcn18/model_140.pth --gpus 0 --dataset vcoco --image_dir images/val2014 --test_with_eval

The loss looks too large, and this is the [log](https://gist.github.com/YueLiao/4790d4fe0774031bfb2f7501ca205d94) which we trained the vcoco based on the old version [code](https://github.com/YueLiao/PPDM/tree/96b4cc3f37cd894318ecf6e22b0a31b540f95950) with DLA-34 to get. But the checkpoint and processed data have missing, I am afraid that it may be caused by the data processed scripts I provided, and it may not be the right version.

**Comment by BestSongEver** [comment]
Thank you very much for helping us find out the cause of the problem.  You are very kind.@YueLiao 
Is there any possibility that you can provide the correct version of the data processed scripts?
Because of my limited computing resources (only one GPU), I have to conduct my experiments on the small dataset V-COCO.
I'd appreciate it if you could help.

**Comment by lylinyang** [comment]
> > > Thinks for your reply. The pre-trained model is "ctdet_coco_resdcn18.pth", and we follow your script to prepare the VCOCO data for inference. I attached the training log for you, please check it if we make something wrong:
> > > [log_train_ori_vcoco_resnet_18_0724.log](https://github.com/YueLiao/PPDM/files/4979724/log_train_ori_vcoco_resnet_18_0724.log)
> > 
> > 
> > The train and test script is as follows:
> > nohup python main.py hoidet --batch_size 256 --master_batch 16 --lr 4.5e-4 --gpus 0,1,2,3,4,5,6,7 --num_workers 16 --load_model ../models/ctdet_coco_resdcn18.pth --arch resdcn_18 --image_dir images/train2014 --dataset vcoco --exp_id hoidet_vcoco_resdcn18 --print_iter 1 >> log_train_ori_vcoco_resnet_18_0724.log 2>&1 &
> > python test_hoi.py hoidet --exp_id hoidet_vcoco_resdcn18 --arch resdcn_18 --load_model ../exp/hoidet/hoidet_vcoco_resdcn18/model_140.pth --gpus 0 --dataset vcoco --image_dir images/val2014 --test_with_eval
> 
> The loss looks too large, and this is the [log](https://gist.github.com/YueLiao/4790d4fe0774031bfb2f7501ca205d94) which we trained the vcoco based on the old version [code](https://github.com/YueLiao/PPDM/tree/96b4cc3f37cd894318ecf6e22b0a31b540f95950) with DLA-34 to get. But the checkpoint and processed data have missing, I am afraid that it may be caused by the data processed scripts I provided, and it may not be the right version.

Thank you very much, compared with mine, your loss log seems normal. Since the scripts you provided may be out-of-date, I wonder if it is convenient for you to provide us the three vcoco data files you processed ("verbcoco_trainval.json", "verbcoco_test.json" and "corre_vcoco.npy") directly via baidu\google drive or email? my email address is 264510914@qq.com.

**Comment by YueLiao** [comment]
@BestSongEver @lylinyang I tried to find these files, and I found the annotation files for vcoco, which can be downloaded from [this](https://drive.google.com/drive/folders/1BDY5p590immmcHhAM55rwaRGXdVIuIJ2?usp=sharing). The trained models are missing.

**Comment by BestSongEver** [comment]
Super cool!  @YueLiao 
I will try the new annotation files.
Thank you again & wish you all the best !

**Comment by lylinyang** [comment]
> @BestSongEver @lylinyang I tried to find these files, and I found the annotation files for vcoco, which can be downloaded from [this](https://drive.google.com/drive/folders/1BDY5p590immmcHhAM55rwaRGXdVIuIJ2?usp=sharing). The trained models are missing.

Thank you for your timely help. We'll try it again.

**Comment by lylinyang** [comment]
@YueLiao We have trained the res18 and dla34 again with the annotations you provided, it works! The mAP is 0.535 and 0.414 for dla34 and res18 respectively. Thank you again for you great help.

**Comment by SherlockHolmes221** [comment]
> @YueLiao We have trained the res18 and dla34 again with the annotations you provided, it works! The mAP is 0.535 and 0.414 for dla34 and res18 respectively. Thank you again for you great help.
**0.535 and 0.414** is get by the offical vcoco evaluation code?
